### ISSUE TITLE
Generating OData 4.0 Queries fails on some entities due to incorrect entity set name

### ISSUE THREAD

**Issue by ZKArunsun** [rappen/FetchXMLBuilder#465]
Generated OData 4.0 Queries seem to be using LogicalCollectionName as the set name of the entities in the url, rather than EntitySetName.

This has no impact most of the time as both these values are equal on almost every entity, but it will cause failure on some specific entities like Sharepoint Document Location:
As per the official [`Entity Reference`](https://docs.microsoft.com/en-us/dynamics365/customerengagement/on-premises/developer/entities/sharepointdocumentlocation) for this entity, the collection name includes a capital P whereas the set name is all lowercase.

**Comment by MarkMpn** [comment]
Good catch, thanks!

**Comment by kaluznyt** [comment]
Is this an issue that when building odata 4.0 for example for queuemembership I see result as:  api/data/v9.2/?$filter=.... ?

**Comment by ZKArunsun** [comment]
@kaluznyt I think this is a separate issue. queuemembership is an intersect entity for a N:N relationship (called queuemembership_association). 
As far as I am aware, there is no way to directly retrieve the intersect entity on N:N association using the OData 4.0 web API, and you should instead query for systemusers?$expand=queuemembership_association if you want to retrieve the the users with their queues, or queues?$expand=queuemembership_association if you want to retrieve the queues and the user within them.

Maybe FetchXMLBuilder could warn the user in that situation, but it is not the same problem and I think you should file another issue for that.

**Comment by MarkMpn** [comment]
I believe this is related - the `LogicalCollectionName` for `queuemembership` is `null`, but the `EntitySetName` is `queuememberships`. An OData query for `/api/data/v9.0/queuememberships` returns data as expected, so the same change should also resolve this issue.

**Comment by ZKArunsun** [comment]
That is right, disregard my previous comment then. I always refer to [Microsoft's Entity References ](https://docs.microsoft.com/en-us/dynamics365/customerengagement/on-premises/developer/entities/queue) for standard entities but as often with Microsoft, even the most complete documentation is incomplete 😄 
Turns out fetching /api/data/v9.1/queuememberships does work even if it is undocumented
